src/core/console.ml

```diff
@@ -221,31 +221,34 @@ let reset_default : unit -> unit = fun () ->
   let reset _ (default, r) = r := default in
   StrMap.iter reset Stdlib.(!boolean_flags)
 
+(** Settings used to compile files. *)
+type state = { st_verbose: int
+             (** Verbosity level. *)
+             ; st_loggers: (char * bool) list
+             (** Loggers enabled. *)
+             ; st_bflags: bool StrMap.t
+             (** Boolean flags. *) }
+
```

Comment:
Nit: wrapping records into a module is common style in some OCaml projects, so that you do
```ocaml
module State = struct
   type t = { verbose : int
                ; ...
```
This has quite a few advantages and the compiler also provides some special support for record scoping.

Reply:
We can even bring `apply_state`, `push_state` &c into the module in that case I guess